File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace BarcodeWithPageInfo
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub xrBarCode1_PrintOnPage(ByVal sender As Object, ByVal e As PrintOnPageEventArgs) Handles xrBarCode1.PrintOnPage
            Dim s As String = "Product Report, Page: " & (e.PageIndex + 1).ToString() & " - Category :" & DirectCast(sender, XRControl).Tag.ToString()
            DirectCast(sender, XRBarCode).BinaryData = GetBytes(s)
        End Sub
        Private Shared Function GetBytes(ByVal str As String) As Byte()
            Dim bytes((str.Length * Len(New Char)) - 1) As Byte
            System.Buffer.BlockCopy(str.ToCharArray(), 0, bytes, 0, bytes.Length)
            Return bytes
        End Function

    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace BarcodeWithPageInfo
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();
        }

        private void xrBarCode1_PrintOnPage(object sender, PrintOnPageEventArgs e)
        {
            string s = "Product Report, Page: " + (e.PageIndex + 1).ToString() + " - Category :" + ((XRControl)sender).Tag.ToString();
            ((XRBarCode)sender).BinaryData = GetBytes(s);
        }
        static byte[] GetBytes(string str) {
            byte[] bytes = new byte[str.Length * sizeof(char)];
            System.Buffer.BlockCopy(str.ToCharArray(), 0, bytes, 0, bytes.Length);
            return bytes;
        }

    }
}
